src/standalone/Selector/BaseSelector.tsx

```diff
@@ -0,0 +1,244 @@
+import React from "react";
+import { TextFieldProps, ListItemText } from "@material-ui/core";
+import { Autocomplete } from "@material-ui/lab";
+import { Add as AddIcon, ExpandMore } from "@material-ui/icons";
+import TextFieldWithHelp, {
+	TextFieldWithHelpProps,
+} from "../UIKit/TextFieldWithHelp";
+import i18n from "../../i18n";
+import { SelectorSmallListItem, SmallListItemIcon } from "../..";
+
+export interface BaseSelectorData {
+	/**
+	 * A unique value
+	 */
+	value: string;
+	/**
+	 * The label to show the user
+	 */
+	label: string;
+	/**
+	 * An optional icon
+	 */
+	icon?: React.ReactNode;
+	/**
+	 * Should the entry be disabled?
+	 */
+	isDisabled?: boolean;
+	/**
+	 * Is this an add new button? (used for special handling in the renderer)
+	 */
+	isAddNewButton?: boolean;
+}
+
+/**
+ * A callback used to get an label value for a specific input (search) value
+ */
+type SelectorLabelCallback = (obj: { inputValue: string }) => string | null;
+
+export interface BaseSelectorProps extends TextFieldWithHelpProps {
+	/**
+	 * Refresh token used to force refreshing data.
+	 * Does not work with truthy multiSelect
+	 */
+	refreshToken?: string;
+	/**
+	 * Data load function
+	 * @param search The user search input
+	 */
+	onLoad: (search: string) => BaseSelectorData[] | Promise<BaseSelectorData[]>;
+	/**
+	 * Selector default options
+	 */
+	defaultOptions?: BaseSelectorData[];
+	/**
+	 * Callback for autocomplete change
+	 */
+	onSelect: (selected: BaseSelectorData | null) => void;
+	/**
+	 * Disable autocomplete control
+	 */
+	disabled?: boolean;
+	/**
+	 * String used for the Autocomplete component
+	 */
+	autocompleteId?: string;
+	/**
+	 * String used to set place holded of the Autocomplete component
+	 */
+	placeholder?: string;
+	/**
+	 * Handler for the "Add new" button
+	 * Add new button only shows if this is set.
+	 */
+	onAddNew?: () => void;
+	/**
+	 * Label for the "Add new" button
+	 */
+	addNewLabel?: string;
+	/**
+	 * The currently selected values
+	 */
+	selected: BaseSelectorData | null;
+	/**
+	 * Enables icons in the list renderers
+	 */
+	enableIcons?: boolean;
+	/**
+	 * Label which is shown if there is no data
+	 */
+	noOptionsText?: string | SelectorLabelCallback;
+	/**
+	 * Label which is shown while loading data
+	 */
+	loadingText?: string | SelectorLabelCallback;
+	/**
+	 * Is the selector clearable?
+	 */
+	disableClearable?: boolean;
+}
+
+const BaseSelector = (props: BaseSelectorProps) => {
+	const {
+		defaultOptions,
+		refreshToken,
+		onSelect,
+		selected,
+		disabled,
+		placeholder,
+		autocompleteId,
+		addNewLabel,
+		onLoad,
+		onAddNew,
+		enableIcons,
+		noOptionsText,
+		loadingText,
+		disableClearable,
+	} = props;
+	const [open, setOpen] = React.useState(false);
```

Comment:
please import all React hooks (useState, useCallback, useMemo, etc) by name.
e.g.
```tsx
import React from "react";

const [state, setState] = React.useState(...)
```
should be
```tsx
import React, { useState } from "react";

const [state, setState] = useState(...);
```